docs/api/implementation-guide.md

```diff
@@ -0,0 +1,80 @@
+|https://pkgs.contoso.com/nuget/v3/feed/registration/|Yes, in a subdirectory of the service index.|
+|https://pkgs.contoso.com/nuget/v3/{guid}/registration/|See below|
+
+In the last example, the server might have a canonical (in this example a guid) name, and have one or more aliases. 
+If the service index request was authenticated on a non-canonical URL (the "friendly" name, in our example `feed`), then no, any requests to resources under the canonical URL will not match `HttpClientHandler`'s rules for `PreAuthenticate`.
+However, if the non-canonical URL is an HTTP redirection to the canonical URL, https://pkgs.contoso.com/nuget/v3/{guid}/index.json, then this URL will be used in `HttpClientHandler`'s credential cache.
+In this case, every request to the service index will have additional latency, due to the redirection.
+
+While NuGet's V3 API was designed to work on a static file server, the search resource is the exception that always requires a dynamic web service to process requests.
+If you wish to host search, or indeed any other NuGet API resource, on different servers, in order to benefit from `HttpClientHandler`'s `PreAuthenticate`, you will need to use a reverse proxy to ensure all customer facing URLs in the service index meet the "same or subdirectory" rule.
+
+## Embedded files
+
+Package [icons](../reference/nuspec.md#icon), [license](../reference/nuspec.md#license), and [readme](../reference/nuspec.md#readme) files can be (and are recommended to be) embedded in the package.
+These files need a URL endpoint, either extracted and put on a static file server, or a URL that dynamically extracts the files from the `.nupkg` on request, so that they can be viewed without downloading the entire `nupkg`.
+If your package repository provides package browsing and viewing package details, you can use the URLs to show customers the embedded content on your website.
+
+Finally, the [package metadata resource](./registration-base-url-resource.md) and [search resource](./search-query-service-resource.md) must contain the hosted URL in the `iconUrl`, `licenseUrl`, and/or `readmeUrl` properties of the JSON response.
+Packages (`.nupkg` files) are immutable and must not be modified.
+
+Note that the license could be an SPDX expression, or an embedded file (but not both).
+Packages that use a license expression, when represented in search and package metadata results, can have the `licenseUrl` set to the license expression, URL encoded, and appended to the end of https://licenses.nuget.org/.
+For example, https://licenses.nuget.org/Apache-2.0.
+
+## Owner field
+
+The [package manifest file (`.nuspec`)](../reference/nuspec.md) have two fields, `<authors>` and `<owners>`.
+Package authors who are packaging third-party content often put the third-party name in the `<authors>` field.
+The `<owners>` field is intended to denote who published the package on a repository, and therefore who should be contacted in case of packing issues or questions.
```

Comment:
Super nitpick since the field is deprecated:

```suggestion
The `<owners>` field was intended to denote who published the package on a repository, and therefore who should be contacted in case of packing issues or questions.
```